pages/detention.vue

```diff
@@ -1,509 +1,427 @@
-                    let item = result.results.bindings[i];
-                    
-                    let wikiPage = {
-                        id: item.item.value,
-                        pageTitle: item.page_title.value,
-                    }
-                    
-                    this.wikiPages.push(wikiPage);
-                }
-                
-                this.loadAccussed();
-            });   
-        },
-        loadAccussed : function() {
-            const sparqlQuery = `SELECT ?item ?itemLabel ?residenceLabel ?residenceCoords ?sexLabel ?link ?occupationLabel ?socialClassificationLabel ?placeOfDeathLabel ?placeOfDeathCoords ?mannerOfDeathLabel ?detentionLocationLabel ?detentionLocationCoords
+       const queryDispatcher = new SPARQLQueryDispatcher( this.sparqlUrl );
+       queryDispatcher.query( sparqlQuery ).then( result => {
+
+         for (let i = 0; i < result.results.bindings.length; i++) {
+           let item = result.results.bindings[i];
+
+           let wikiPage = {
+             id: item.item.value,
+             pageTitle: item.page_title.value,
+           }
+
+           this.wikiPages.push(wikiPage);
+         }
+
+         this.loadAccussed();
+       });
+     },
+     loadAccussed: function () {
+
+       const sparqlQuery = `SELECT distinct ?item ?itemLabel ?investigationDate
+            ?residenceLabel ?residenceCoords ?sexLabel ?link ?occupationLabel ?socialClassificationLabel
+            ?placeOfDeathLabel ?placeOfDeathCoords ?mannerOfDeathLabel ?detentionLocationLabel ?detentionLocationCoords
             WHERE
             {
               ?item wdt:P4478 ?witch .
-              optional {
-                ?item wdt:P551 ?residence .
-                ?residence wdt:P625 ?residenceCoords .
-              }
+              ?item wdt:P551 ?residence .
+              ?residence wdt:P625 ?residenceCoords .
               optional { ?item wdt:P21 ?sex } .
               ?item wdt:P4478 ?link .
-              optional { ?item wdt:P106 ?occupation}
-              optional { ?item wdt:P3716 ?socialClassification}
+              optional { ?item wdt:P106 ?occupation .}
+              optional { ?item wdt:P3716 ?socialClassification .}
               optional {
                 ?item wdt:P20 ?placeOfDeath .
-                ?placeOfDeath wdt:P625 ?placeOfDeathCoords
+                ?placeOfDeath wdt:P625 ?placeOfDeathCoords .}
+              optional { ?item wdt:P1196 ?mannerOfDeath .}
+              optional { ?item p:P793 ?significantEventStatement .
+              ?significantEventStatement ps:P793 wd:Q66458810 .
+              OPTIONAL {?significantEventStatement pq:P585 ?investigationPoint }.
+              OPTIONAL {?significantEventStatement pq:P580 ?investigationStart }
               }
-              optional { ?item wdt:P1196 ?mannerOfDeath}
+              BIND(IF(BOUND(?investigationPoint), ?investigationPoint, ?investigationStart) as ?investigationDate)
```

Comment:
Just a question. What is  BIND(IF(BOUND(?investigationPoint), ?investigationPoint, ?investigationStart) as ?investigationDate) doing?

Sorry totally forgot how sparql works